### PR TITLE
implement branding support

### LINKED ISSUE #3972: Add logo/branding support in core
Maybe consider adding logo/branding support as a core feature. It is impossible to create a separate admin theme that will use the main theme logo without hardcoding it. 
Branding is some sort of a central feature similar to site name and description. It could be used as well as a default image for opengraph in modules like news articles etc...

### PR DESCRIPTION
**Pull request by Guite**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | no
| Deprecations?     | no
| Fixed tickets     | fixed #3972 
| Refs tickets      | -
| License           | MIT
| Changelog updated | yes

## Description

This PR is going to add the required image handling and further functionality for providing remaining/missing branding capabilities.

I have decided not to integrate realfavicongenerator directly to keep things simple and avoid complexity while keeping flexible customisation possible at the same time.

Screenshot_20200412_162204

### CODE REVIEW

File: src/system/ThemeModule/Resources/views/Config/config.html.twig [hunk 1]
```diff
@@ -18,4 +18,66 @@
     </div>
 </div>
 {{ form_end(form) }}
+{% macro imageRow(name, label, description, code, isImage = true) %}
+    {% set useImageFunction = name in ['logo', 'mobileLogo', 'icon'] %}
+    {% set path = useImageFunction ? siteImagePath(code) : asset(code) %}
+    <tr>
+        <th id="h{{ name|capitalize }}" scope="row" headers="hType">{{ label }}</th>
+        <td headers="h{{ name|capitalize }} hDescription">{{ description }}</td>
+        <td headers="h{{ name|capitalize }} hPath">{{ path }}</td>
```
Comment (craigh) [inline src/system/ThemeModule/Resources/views/Config/config.html.twig:27]: maybe do not allow text wrapping in the path column. I was confused at first when I looked at the graphic

Comment (Guite) [reply src/system/ThemeModule/Resources/views/Config/config.html.twig:27]: then the table needs horizontal scrolling which I wanted to avoid

**Comment by Guite** [comment]
added docs in 406e146a84093b7a98642a5a96a6363ba320c5ea

**Comment by Guite** [comment]
added docs in 673efe700d632187e2aa862b5c4a4900cd9260a9

**Pull request merged**